src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +49,46 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static ClassifiedTextElement BuildClassifiedTextElementForClassifiedSpans(IEnumerable<ClassifiedSpan> classifiedSpans, ITextSnapshot snapshot, CancellationToken cancellationToken)
+        {
+            Contract.Assert(classifiedSpans != null && classifiedSpans.Any());
+
+            // Convert spans to textruns
+            var runs = new List<ClassifiedTextRun>();
+            var lastSpanEnd = -1;
+            foreach (var span in classifiedSpans)
+            {
+                // Add whitespace
+                if (lastSpanEnd > 0 && span.TextSpan.Start > lastSpanEnd)
+                {
+                    // find the gap between last span and current span, read the gap text from document
+                    var spanGap = new Span(lastSpanEnd, span.TextSpan.Start - lastSpanEnd);
+                    var spanText = snapshot.GetText(spanGap);
+
+                    // special handling for new line, remove all the whitespace at the begining of the new line before the open brace "{"
+                    if (span.ClassificationType == ClassificationTypeNames.Punctuation)
+                    {
+                        if (spanText.StartsWith("\r\n"))
+                        {
+                            spanText = "\r\n";
+                        }
+                        else if (spanText.StartsWith("\n"))
+                        {
+                            // new line in linux
+                            spanText = "\n";
+                        }
+                    }
+
+                    runs.Add(new ClassifiedTextRun(ClassificationTypeNames.WhiteSpace, spanText));
+                }
+
+                lastSpanEnd = span.TextSpan.End;
+                runs.Add(new ClassifiedTextRun(span.ClassificationType, snapshot.GetText(span.TextSpan.ToSpan())));
```

Comment:
should we have a test to make sure our classificationType doesn't change? since we give ours as it is to VS?

Reply:
@heejaechang  Do you mean adding a test to make sure the members of this class does not change? https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Classification/ClassificationTypeNames.cs

Reply:
since both are public API, might be okay. worried since this assumes names are same between 2 different apis.